### PR TITLE
[NEURON][TF][BUILD] - Update TF 2.10 image with latest PIP packages versions

### PR DESCRIPTION
**Pull request by catalin-manciu-aws**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/inference/docker/2.10/py3/sdk2.8.0/Dockerfile.neuron [hunk 1]
```diff
@@ -58,12 +58,12 @@ RUN apt-get update \
...
  && apt-get -y install --no-install-recommends \
-    nginx=1.20.1* \
-    nginx-module-njs=1.20.1* \
+    nginx=1.22.1* \
+    nginx-module-njs=1.22.1* \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*
 
-# Install python3.9
+# Install Python3
```
Comment (aws-ceenugal) [inline tensorflow/inference/docker/2.10/py3/sdk2.8.0/Dockerfile.neuron:66]: Check : why is this change needed?

Comment (catalin-manciu-aws) [reply tensorflow/inference/docker/2.10/py3/sdk2.8.0/Dockerfile.neuron:66]: We're installing Python 3.8 with this image. I just wanted to keep it more generic to minimize future updates, I'll change it to the actual version number

File: tensorflow/inference/docker/2.10/py3/sdk2.8.0/Dockerfile.neuron [hunk 2]
```diff
@@ -76,7 +76,7 @@ RUN wget -qO - https://apt.repos.neuron.amazonaws.com/GPG-PUB-KEY-AMAZON-AWS-NEU
 RUN apt-get update
 
 RUN apt-get install -y \
-    tensorflow-model-server-neuron=2.8.* \
+    tensorflow-model-server-neuron=1.15.0.2.3.0.0 \
```
Comment (aws-ceenugal) [inline tensorflow/inference/docker/2.10/py3/sdk2.8.0/Dockerfile.neuron:79]: Does this need to be 2.10?

Comment (catalin-manciu-aws) [reply tensorflow/inference/docker/2.10/py3/sdk2.8.0/Dockerfile.neuron]: Updated with the latest tensorflow 2.10-compatible release (which is now neuronx but still compatible)

File: test/dlc_tests/sanity/test_safety_check.py [hunk 3]
```diff
@@ -461,6 +461,24 @@
...
+                "53857", # TF 1.15.5, 2.10
+                "53856", # TF 1.15.5, 2.10
+                "53854", # TF 1.15.5, 2.10
+                "53852", # TF 1.15.5, 2.10
+                "53851", # TF 1.15.5, 2.10
+                "53849", # TF 1.15.5, 2.10
+                "53848", # TF 1.15.5, 2.10
+                "53847", # TF 1.15.5, 2.10
+                "53845", # TF 1.15.5, 2.10
+                "53846", # TF 1.15.5, 2.10
```
Comment (shantanutrip) [inline test/dlc_tests/sanity/test_safety_check.py:481]: These vulns need to be added to https://github.com/aws/deep-learning-containers/blob/master/data/ignore_ids_safety_scan.json#L64. You can remove this from here altogether in case required, but put it over at the [location](https://github.com/aws/deep-learning-containers/blob/master/data/ignore_ids_safety_scan.json#L64) I mentioned above.

**Review by aws-ceenugal — APPROVED** [review]

File: test/dlc_tests/sanity/test_safety_check.py (hunk 3 above)
Comment (catalin-manciu-aws) [reply test/dlc_tests/sanity/test_safety_check.py:481]: Done

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py [hunk 4]
```diff
@@ -164,8 +165,13 @@ def run_ec2_tensorflow_inference(image_uri, ec2_connection, grpc_port, region, t
             )
         if telemetry_mode:
             check_telemetry(ec2_connection, container_name)
+    except:
+        inference_test_failed = True
+        remote_out = ec2_connection.run(f"docker logs {container_name}")
```
Comment (shantanutrip) [inline test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py:170]: Should we use `hide=True` and `warn=True` over here?

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py [hunk 5]
```diff
@@ -27,7 +27,7 @@
 @pytest.mark.model("mnist")
 @pytest.mark.parametrize("ec2_instance_type", TF_EC2_NEURON_ACCELERATOR_TYPE, indirect=True)
 # FIX ME: Sharing the AMI from neuron account to DLC account; use public DLAMI with inf1 support instead
```
[the remark this answers is not in the document]
Comment (catalin-manciu-aws) [reply test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py]: Removed

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py (hunk 4 above)
Comment (catalin-manciu-aws) [reply test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py]: Updated in latest commit

File: tensorflow/inference/docker/2.10/py3/sdk2.8.0/Dockerfile.neuron [hunk 6]
```diff
@@ -76,7 +76,7 @@ RUN wget -qO - https://apt.repos.neuron.amazonaws.com/GPG-PUB-KEY-AMAZON-AWS-NEU
 RUN apt-get update
 
 RUN apt-get install -y \
-    tensorflow-model-server-neuron=2.8.* \
+    tensorflow-model-server-neuronx=2.10.1.2.6.5.0 \
```
[the remark this answers is not in the document]
Comment (shantanutrip) [reply tensorflow/inference/docker/2.10/py3/sdk2.8.0/Dockerfile.neuron:79]: Got it, thanks!!

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (shantanutrip) [reply test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py]: Thanks!!
